### PR TITLE
feat: assign lecture start to event creation time

### PR DESCRIPTION
**Pull request by LucasFalkowsky**
- closes https://github.com/corona-school/project-user/issues/1147

## What was done:

- refactoring of bucket creation für lecture related buckets
- additional context for meeting join actions
- lecture start as date for joined meeting events

### CODE REVIEW

File: common/achievement/bucket.ts
Comment (dhenkel92) [inline common/achievement/bucket.ts:1]: 🤔 This change looks suspicious

**Review by dhenkel92 — COMMENTED** [review]
We had the same problem for the notification system, why not taking the same solution?

File: common/achievement/bucket.ts
Comment (LucasFalkowsky) [reply common/achievement/bucket.ts:1]: what do you mean, suspicious? Filtering is no longer necessary, when setting the event timestamp to lecture.start.

But If you want to keep this, we can.

Comment (dhenkel92) [reply common/achievement/bucket.ts:1]: I wasn't expecting any changes here 😅 They seemed unrelated and more like merge errors.
Fine for me to remove that part 👍

**Pull request merged**